File: src/main/java/main/MessageChecker.java
package main;

import util.Message;

import java.io.UnsupportedEncodingException;

/**
 * Created by Eldath Ray on 2017/3/28.
 *
 * @author Eldath Ray
 */
public class MessageChecker {
    public static boolean checkEncode(Message message) {
        String content = message.getContent().toLowerCase();
        try {
            if (!content.equals(new String(content.getBytes("GB2312"), "GB2312"))) {
                message.response("@" + message.getSenderNickName() + " 您消息的编码好像不对劲啊╮(╯_╰)╭");
                return false;
            }
        } catch (UnsupportedEncodingException ignore) {
        }
        return true;
    }
}

File: src/main/kotlin/avalon/main/MessageChecker.kt
package avalon.main

import avalon.api.Flag.AT
import avalon.tool.pool.Constants.Basic.LANG
import avalon.util.Message
import java.nio.charset.Charset

object MessageChecker {
	@JvmStatic
	fun check(message: Message): Boolean = checkEncode(message)

	@JvmStatic
	private fun checkEncode(message: Message): Boolean {
		val content = message.content
		val charset = Charset.forName("GB2312")
		if (!(content.contentEquals(String(content.toByteArray(charset), charset)))) {
			message.response("${AT(message.senderUid)} ${LANG.getString("group.handler.wrong_encoding")}")
			return false
		}
		return true
	}
}